File: electit_backend/src/main/java/org/sami/electit/shared/domain/entities/Candidate.java
package org.sami.electit.shared.domain.entities;

import org.sami.electit.features.candidates.shared.api.dtos.CandidateDTO;
import org.sami.electit.features.candidates.shared.api.dtos.CandidateInput;
import org.springframework.data.neo4j.core.schema.*;

@Node
public record Candidate(
		@Id @GeneratedValue Long id,
		String name,
		String party,
		String manifesto,
		String pictureUrl
) {
	public static Candidate fromDTO(CandidateInput candidateInput) {
		return new Candidate(null, candidateInput.name(), candidateInput.party(), candidateInput.manifesto(), candidateInput.pictureUrl());
	}

	public CandidateDTO toDTO(Integer votesCount) {
		return new CandidateDTO(id(), name(), party(), manifesto(), pictureUrl(), votesCount);
	}
}

File: electit_frontend/lib/features/shared/domain/entities/candidate.dart
class Candidate {
  final String id;
  final String name;
  final String party;
  final String manifesto;
  final String? pictureUrl;
  final int votesCount;
  final bool hasMyVote;

  Candidate({
    required this.id,
    required this.name,
    required this.party,
    required this.manifesto,
    required this.pictureUrl,
    required this.votesCount,
    this.hasMyVote = false,
  });

  double votesPercentage(int electionVotesCount) {
    return votesCount / electionVotesCount;
  }
}
